### PR TITLE
Move destructor into invalidate

### PR DESCRIPTION
**Pull request by magicismight**
# Summary

WebView instance won't release properly after React component been unmounted, and the `[NSNotificationCenter defaultCenter]` observers are never been removed.

changes:

* I have moved `[NSNotificationCenter defaultCenter]` observer to didMoveToWindow, it is unnecessary to add those observers in initWithFrame.
* remove dealloc, the `[NSNotificationCenter defaultCenter]` observers should be removed before dealloc, or the `dealloc` will never be called.
* move `removeFromSuperview`'s code into `invalidate`, somehow the `removeFromSuperview` is not been called while the React component is unmounted, `RCTInvalidating` protocol is solid in this case.

## Test Plan

Mount a WebView React component on screen and unmount the React component after a while, check if the WebView instance can be accessed from a Safari remote debugger.

### What's required for testing (prerequisites)?

A Safari remote debugger is required, it will display the valid WebViews in simulator.

### What are the steps to reproduce (after prerequisites)?

An unmounted WebView instance can be access by Safari remote debugger.

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅   |
| Android |    ❌     |

## Checklist

- [x]  I have tested this on a device and a simulator
- [ ] I added the documentation in `README.md`
- [ ] I mentioned this change in `CHANGELOG.md`
- [ ] I updated the typed files (TS and Flow)
- [ ] I added a sample use of the API in the example project (`example/App.js`)

### CODE REVIEW

File: android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java [hunk 1]
```diff
@@ -420,6 +420,16 @@ public void setIncognito(WebView view, boolean enabled) {
...
+    // INFO: ensure all other props has been set
+    view.post(new Runnable() {
+      @Override
+      public void run() {
+        loadSource(view, source);
+      }
+    });
+  }
+
+  private void loadSource(WebView view, @Nullable ReadableMap source) {
```
Comment (jamonholmgren) [inline android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java]: Could you submit this in a different PR, since this PR is about iOS?

**Comment by safaiyeh** [comment]
@magicismight any update on separating this PR?
